shell/platform/windows/keyboard_manager_win32.h

```diff
@@ -92,30 +104,40 @@ class KeyboardManagerWin32 {
                      LPARAM const lparam);
 
  private:
+  struct Win32Message {
+    UINT const action;
+    WPARAM const wparam;
+    LPARAM const lparam;
+
+    bool IsHighSurrogate() const { return IS_HIGH_SURROGATE(wparam); }
+
+    bool IsLowSurrogate() const { return IS_LOW_SURROGATE(wparam); }
+
+    bool IsGeneralKeyDown() const {
+      return action == WM_KEYDOWN || action == WM_SYSKEYDOWN;
+    }
+  };
+
   struct PendingEvent {
-    uint32_t key;
+    uint16_t key;
```

Comment:
Could you use the Windows type here (i.e. SHORT), and for the others that are derived from Windows struct values?

Reply:
At some point we have to convert the Win32 types to fixed-width types. My idea here is that, the members of `Win32Message` preserve the original value, since they are to be redispatched, while the members of `PendingEvent` should be fixed-width values, since they are processed values and will be send to `OnKey` of the platform-independent `FlutterWindowsView`.

Reply:
I agree that the change here (or most of the numerical types here) is debatable. `WPARAM` is technically uint64 nowadays in a 64-bit system, but the API dated back to the 32-bit era, and practically the virtual code never exceeds 16 bit.

Reply:
Sure, and I'm sure that practically it makes no difference, but from a correctness standpoint it would be better to use the system defined types instead of making the (yes, perfectly valid) assumptions about the sizes of the types.

I'm OK with making the assumptions, though, if you think that the documentation value of showing the type sizes is important enough to trump the correctness.

Reply:
It might be sufficient to have a compile-time `static_assert` that `sizeof(SHORT) == sizeof(uint16_t)`.

Reply:
I've replaced the types of `key` and `action` to win32 types since they directly come from win32 values.